test/testdata/infer/attached_class_self_new.rb

```diff
@@ -0,0 +1,91 @@
+# typed: true
+
+class NoArgs
+  extend T::Sig
+
+  sig {returns(T.experimental_attached_class)}
+  def self.make
+    T.reveal_type(self.new) # error: Revealed type: `T.class_of(NoArgs)::<AttachedClass>`
```

Comment:
Might be worth adding a test with an implicit self too

Reply:
Good point!